### PR TITLE
Refactor/user-data-class

### PR DESCRIPTION
**Pull request by RMartin0717**
# Pull Request Template
## Description

I used the original project spec to expand on testing to provide better testing coverage of the User class functionality. Built out User class methods missing from code base to add to recipesToCook and fix searchFavoritesByTag functionality.

Fixes # (issue)
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactoring

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/user.js [hunk 1]
```diff
@@ -18,17 +19,23 @@ class User {
...
+      return acc.concat(checkTag)
+    }, []);
+    return searchResults
   }
 
-  findFavorites(strgToSrch) {
+  searchFavoritesByNameOrIng(searchWord) {
     return this.favoriteRecipes.filter(recipe => {
-      return recipe.name.includes(strgToSrch)
+      return recipe.name.includes(searchWord)
```
Comment (bhohnco) [inline src/user.js:36]: thank you for fixing this, that argument was bothering me

File: test/user-test.js [hunk 2]
```diff
@@ -21,39 +20,194 @@ describe('User', () => {
...
   });
-
-  it('Should inform User if they lack required ingredients for a given recipe', () => {
-    expect(user1.checkPantry(recipeIngredients)).to.eql(missingIngredientsWithPrice);
+  it('Should be able to search favoriteRecipes by name or ingredient', () => {
+    user1.addToFavorites(recipeData[0]);
+    user1.addToFavorites(recipeData[1]);
+    user1.addToFavorites(recipeData[2]);
+    const searchByIng = user1.searchFavoritesByNameOrIng("brown sugar");
+    expect(searchByIng).to.deep.equal([recipeData[2]]);
```
Comment (bhohnco) [inline test/user-test.js:209]: testing is looking good, nice job

**Review by bhohnco — APPROVED** [review]

File: src/user.js [hunk 3]
```diff
@@ -18,17 +19,23 @@ class User {
...
+  searchFavoritesByTag(searchTags) {
+    const searchResults = searchTags.reduce((acc, tag) => {
+      const checkTag = this.favoriteRecipes.filter(recipe => recipe.tags.includes(tag));
+      return acc.concat(checkTag)
+    }, []);
+    return searchResults
   }
 
-  findFavorites(strgToSrch) {
+  searchFavoritesByNameOrIng(searchWord) {
```
Comment (michannstoner) [inline src/user.js:34]: Agreed with Bryan, thank you!

**Review by michannstoner — APPROVED** [review]
Looks good, thanks for your work on this Riley!

**Pull request merged**